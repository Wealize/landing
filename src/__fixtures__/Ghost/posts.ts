import { PostOrPage } from '@tryghost/content-api'

export const WITHOUT_TAGS: PostOrPage = {
  id: '6017df3f6b0d7a001c15cc7c',
  uuid: 'b4283683-b51e-4c67-ba42-66de093b1e05',
  title: 'Writing posts with Ghost ✍️',
  slug: 'the-editor',
  html: "<h2 id=\"just-start-writing\">Just start writing</h2><p>Ghost has a powerful visual editor with familiar formatting options, as well as the ability to add dynamic content.</p><p>Select your text to add formatting such as headers or to create links. Or use Markdown shortcuts to do the work for you - if that's your thing. </p><figure class=\"kg-card kg-image-card\"><img src=\"https://static.ghost.org/v2.0.0/images/formatting-editor-demo.gif\" class=\"kg-image\"></figure><h2 id=\"rich-editing-at-your-fingertips\">Rich editing at your fingertips</h2><p>The editor can also handle rich media objects, called <strong>cards</strong>, which can be organised and re-ordered using drag and drop. </p><p>You can insert a card either by clicking the  <code>+</code>  button, or typing  <code>/</code>  on a new line to search for a particular card. This allows you to efficiently insert<strong> images</strong>, <strong>markdown</strong>, <strong>html, embeds </strong>and more.</p><p><strong>For example</strong>:</p><ul><li>Insert a video from YouTube directly by pasting the URL</li><li>Create unique content like buttons or forms using the HTML card</li><li>Need to share some code? Embed code blocks directly </li></ul><pre><code>&lt;header class=\"site-header outer\"&gt;\n    &lt;div class=\"inner\"&gt;\n        {{&gt; \"site-nav\"}}\n    &lt;/div&gt;\n&lt;/header&gt;</code></pre><p>It's also possible to share links from across the web in a visual way using bookmark cards that automatically render information from a websites meta data. Paste any URL to try it out: </p><figure class=\"kg-card kg-bookmark-card\"><a class=\"kg-bookmark-container\" href=\"https://ghost.org\"><div class=\"kg-bookmark-content\"><div class=\"kg-bookmark-title\">Ghost: The #1 open source headless Node.js CMS</div><div class=\"kg-bookmark-description\">The world’s most popular modern open source publishing platform. A headless Node.js CMS used by Apple, Sky News, Tinder and thousands more. MIT licensed, with 30k+ stars on Github.</div><div class=\"kg-bookmark-metadata\"><img class=\"kg-bookmark-icon\" src=\"https://ghost.org/icons/icon-512x512.png?v=188b8b6d743c6338ba2eab2e35bab4f5\"><span class=\"kg-bookmark-publisher\">Ghost</span></div></div><div class=\"kg-bookmark-thumbnail\"><img src=\"https://ghost.org/images/meta/Ghost.png\"></div></a></figure><h2 id=\"working-with-images-in-posts\">Working with images in posts</h2><p>You can add images to your posts in many ways:</p><ul><li>Upload from your computer</li><li>Click and drag an image into the browser</li><li>Paste directly into the editor from your clipboard</li><li>Insert using a URL</li></ul><h3 id=\"image-sizes\">Image sizes</h3><p>Once inserted you can blend images beautifully into your content at different sizes and add captions and alt tags wherever needed.</p><figure class=\"kg-card kg-image-card\"><img src=\"https://static.ghost.org/v3.0.0/images/image-sizes-ghost-editor.png\" class=\"kg-image\"></figure><h3 id=\"image-galleries\">Image galleries</h3><p>Tell visual stories using the gallery card to add up to 9 images that will display as a responsive image gallery: </p><figure class=\"kg-card kg-gallery-card kg-width-wide\"><div class=\"kg-gallery-container\"><div class=\"kg-gallery-row\"><div class=\"kg-gallery-image\"><img src=\"https://static.ghost.org/v3.0.0/images/gallery-sample-1.jpg\" width=\"6000\" height=\"4000\"></div><div class=\"kg-gallery-image\"><img src=\"https://static.ghost.org/v3.0.0/images/gallery-sample-2.jpg\" width=\"5746\" height=\"3831\"></div><div class=\"kg-gallery-image\"><img src=\"https://static.ghost.org/v3.0.0/images/gallery-sample-3.jpg\" width=\"5872\" height=\"3915\"></div></div></div></figure><h3 id=\"image-optimisation\">Image optimisation</h3><p>Ghost will automatically resize and optimise your images with lossless compression. Your posts will be fully optimised for the web without any extra effort on your part.</p><h2 id=\"next-publishing-options\">Next: Publishing Options</h2><p>Once your post is looking good, you'll want to use the <a href=\"https://wlz-blog.herokuapp.com/publishing-options/\">publishing options</a> to ensure it gets distributed in the right places, with custom meta data, feature images and more.</p>",
  comment_id: '6017df3f6b0d7a001c15cc7c',
  feature_image: 'https://static.ghost.org/v3.0.0/images/writing-posts-with-ghost.png',
  featured: false,
  created_at: '2021-02-01T11:00:15.000+00:00',
  updated_at: '2021-02-01T11:00:15.000+00:00',
  published_at: '2021-02-01T11:00:20.000+00:00',
  custom_excerpt: 'Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.',
  codeinjection_head: null,
  codeinjection_foot: null,
  custom_template: null,
  canonical_url: null,
  url: 'https://wlz-blog.herokuapp.com/the-editor/',
  excerpt: 'Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.',
  reading_time: 3,
  og_image: null,
  og_title: null,
  og_description: null,
  twitter_image: null,
  twitter_title: null,
  twitter_description: null,
  meta_title: null,
  meta_description: null
}

export const CLIENT_STORIES: PostOrPage[] = [
  {
    id: '6017df3f6b0d7a001c15cc7c',
    uuid: 'b4283683-b51e-4c67-ba42-66de093b1e05',
    title: 'Writing posts with Ghost ✍️',
    slug: 'the-editor',
    tags: [
      {
        id: '602112e0638f28001c9994e3',
        name: 'Client Story',
        slug: 'client-story',
        description: null,
        feature_image: null,
        visibility: 'public',
        meta_title: null,
        meta_description: null,
        url: 'https://wlz-blog.herokuapp.com/tag/client-story/'
      }
    ],
    html: '<p>test html</p>',
    comment_id: '6017df3f6b0d7a001c15cc7c',
    feature_image: 'https://static.ghost.org/v3.0.0/images/writing-posts-with-ghost.png',
    featured: false,
    created_at: '2021-02-01T11:00:15.000+00:00',
    updated_at: '2021-02-01T11:00:15.000+00:00',
    published_at: '2021-02-01T11:00:20.000+00:00',
    custom_excerpt: 'Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.',
    codeinjection_head: null,
    codeinjection_foot: null,
    custom_template: null,
    canonical_url: null,
    url: 'https://wlz-blog.herokuapp.com/the-editor/',
    excerpt: 'Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.',
    reading_time: 3,
    og_image: null,
    og_title: null,
    og_description: null,
    twitter_image: null,
    twitter_title: null,
    twitter_description: null,
    meta_title: null,
    meta_description: null
  },
  {
    id: '6017df3f6b0d7a001c15cc8c',
    uuid: 'b4283683-b51e-4c67-ba42-66de093b1e05',
    title: 'Writing posts with Ghost ✍️ 2',
    slug: 'the-editor',
    tags: [
      {
        id: '602112e0638f28001c9994e3',
        name: 'Client Story',
        slug: 'client-story',
        description: null,
        feature_image: null,
        visibility: 'public',
        meta_title: null,
        meta_description: null,
        url: 'https://wlz-blog.herokuapp.com/tag/client-story/'
      }
    ],
    html: '<p>test html</p>',
    comment_id: '6017df3f6b0d7a001c15cc7c',
    feature_image: 'https://static.ghost.org/v3.0.0/images/writing-posts-with-ghost.png',
    featured: false,
    created_at: '2021-02-01T11:00:15.000+00:00',
    updated_at: '2021-02-01T11:00:15.000+00:00',
    published_at: '2021-02-01T11:00:20.000+00:00',
    custom_excerpt: 'Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.',
    codeinjection_head: null,
    codeinjection_foot: null,
    custom_template: null,
    canonical_url: null,
    url: 'https://wlz-blog.herokuapp.com/the-editor/',
    excerpt: 'Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.',
    reading_time: 3,
    og_image: null,
    og_title: null,
    og_description: null,
    twitter_image: null,
    twitter_title: null,
    twitter_description: null,
    meta_title: null,
    meta_description: null
  }
]

export const CLIENT_STORY:PostOrPage = {
  id: '602112b7638f28001c9994dc',
  uuid: 'eaa52af4-ca2e-46b1-a471-ddde428631c3',
  title: 'Test client story post',
  slug: 'test-client-storie-post',
  html: '<p>Test client storie post</p>',
  comment_id: '602112b7638f28001c9994dc',
  feature_image: null,
  featured: false,
  created_at: '2021-02-08T10:30:15.000+00:00',
  updated_at: '2021-02-08T10:30:55.000+00:00',
  published_at: 'Monday, February 8, 2021',
  custom_excerpt: null,
  codeinjection_head: null,
  codeinjection_foot: null,
  custom_template: null,
  canonical_url: null,
  tags: [
    {
      id: '602112e0638f28001c9994e3',
      name: 'Client Story',
      slug: 'client-story',
      description: null,
      feature_image: null,
      visibility: 'public',
      meta_title: null,
      meta_description: null,
      url: 'https://wlz-blog.herokuapp.com/tag/client-story/'
    }
  ],
  primary_tag: {
    id: '602112e0638f28001c9994e3',
    name: 'Client Story',
    slug: 'client-story',
    description: null,
    feature_image: null,
    visibility: 'public',
    meta_title: null,
    meta_description: null,
    url: 'https://wlz-blog.herokuapp.com/tag/client-story/'
  },
  url: 'https://wlz-blog.herokuapp.com/test-client-storie-post/',
  excerpt: 'Test client storie post',
  reading_time: 0,
  og_image: null,
  og_title: null,
  og_description: null,
  twitter_image: null,
  twitter_title: null,
  twitter_description: null,
  meta_title: null,
  meta_description: null
}

export const NEWS:PostOrPage = {
  id: '602112b7638f28001c9994dc',
  uuid: 'eaa52af4-ca2e-46b1-a471-ddde428631c3',
  title: 'Test client story post',
  slug: 'test-news-post',
  html: '<p>Test news post</p>',
  comment_id: '602112b7638f28001c9994dc',
  feature_image: null,
  featured: false,
  created_at: '2021-02-08T10:30:15.000+00:00',
  updated_at: '2021-02-08T10:30:55.000+00:00',
  published_at: 'Monday, February 8, 2021',
  custom_excerpt: null,
  codeinjection_head: null,
  codeinjection_foot: null,
  custom_template: null,
  canonical_url: null,
  tags: [
    {
      id: '602112e0638f28001c9994e5',
      name: 'News',
      slug: 'news',
      description: null,
      feature_image: null,
      visibility: 'public',
      meta_title: null,
      meta_description: null,
      url: 'https://wlz-blog.herokuapp.com/tag/news/'
    }
  ],
  primary_tag: {
    id: '602112e0638f28001c9994e5',
    name: 'News',
    slug: 'news',
    description: null,
    feature_image: null,
    visibility: 'public',
    meta_title: null,
    meta_description: null,
    url: 'https://wlz-blog.herokuapp.com/tag/news/'
  },
  url: 'https://wlz-blog.herokuapp.com/test-news-post/',
  excerpt: 'Test news post',
  reading_time: 0,
  og_image: null,
  og_title: null,
  og_description: null,
  twitter_image: null,
  twitter_title: null,
  twitter_description: null,
  meta_title: null,
  meta_description: null
}

export const EXTERNAL_CONTENT:PostOrPage = {
  id: '602112b7638f28001c9994dc',
  uuid: 'eaa52af4-ca2e-46b1-a471-ddde428631c3',
  title: 'Test external content post',
  slug: 'test-external-content-post',
  html: '<figure class="kg-card kg-bookmark-card"><a class="kg-bookmark-container" href="https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/String/match"><div class="kg-bookmark-content"><div class="kg-bookmark-title">String.prototype.match() - JavaScript | MDN</div><div class="kg-bookmark-description">El método match() se usa para obtener todas las ocurrencias de una expresión regular dentro de una cadena.</div><div class="kg-bookmark-metadata"><img class="kg-bookmark-icon" src="https://developer.mozilla.org/favicon144.png"><span class="kg-bookmark-publisher">MDN</span></div></div></a></figure>',
  comment_id: '602112b7638f28001c9994dc',
  feature_image: null,
  featured: false,
  created_at: '2021-02-08T10:30:15.000+00:00',
  updated_at: '2021-02-08T10:30:55.000+00:00',
  published_at: 'Monday, February 8, 2021',
  custom_excerpt: null,
  codeinjection_head: null,
  codeinjection_foot: null,
  custom_template: null,
  canonical_url: null,
  tags: [
    {
      id: '602112e0638f28001c9994e5',
      name: 'External Content',
      slug: 'external-content',
      description: null,
      feature_image: null,
      visibility: 'public',
      meta_title: null,
      meta_description: null,
      url: 'https://wlz-blog.herokuapp.com/tag/external-content/'
    }
  ],
  primary_tag: {
    id: '602112e0638f28001c9994e5',
    name: 'External Content',
    slug: 'external-content',
    description: null,
    feature_image: null,
    visibility: 'public',
    meta_title: null,
    meta_description: null,
    url: 'https://wlz-blog.herokuapp.com/tag/external-content/'
  },
  url: 'https://wlz-blog.herokuapp.com/test-news-post/',
  excerpt: 'Test external content post',
  reading_time: 0,
  og_image: null,
  og_title: null,
  og_description: null,
  twitter_image: null,
  twitter_title: null,
  twitter_description: null,
  meta_title: null,
  meta_description: null
}
