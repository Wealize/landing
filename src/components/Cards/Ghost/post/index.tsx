import React from 'react'
import Link from 'next/link'

import GhostPost from '../../../../interfaces/Ghost/GhostPost'

import { Container, CardHeadline, CardPublishedAt } from './styles'

interface GhostPostCardProps {
  post: GhostPost
}

const GhostPostCard = (props: GhostPostCardProps) => {
  const { post } = props
  return (
    <Link href="/news-room/[slug]" as={`/news-room/${post.slug}`} passHref>
      <Container className="ghost-post-card">
      <CardHeadline>{post.title}</CardHeadline>
      <CardPublishedAt>{post.published_at}</CardPublishedAt>
    </Container>
    </Link>
  )
}

export default React.memo(GhostPostCard)
