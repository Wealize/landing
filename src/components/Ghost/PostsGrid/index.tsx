import { PostOrPage } from '@tryghost/content-api'
import useTranslation from 'next-translate/useTranslation'
import Link from 'next/link'
import React from 'react'
import ReactPlaceholder from 'react-placeholder'

import GhostPostCard from '../../../components/Ghost/PostCard'

import { Container, SectionTitle, ContainerPosts, ShowMoreLink } from './styles'


type GhostPostsGridProps = {
  posts: PostOrPage[]
  sectionTitle?: string,
  maxPaginationPages?: number,
  coverSectionPageName?: string
}

const GhostPostsGrid = (props: GhostPostsGridProps) => {
  const {
    posts,
    sectionTitle,
    maxPaginationPages,
    coverSectionPageName
  } = props
  const { t } = useTranslation('common')

  return (
    <ReactPlaceholder
        type='media'
        ready={posts?.length > 0}
        style={{
          marginTop: '7rem',
          width: '100%',
          height: '100%'
        }}>
      <Container>
        {
          posts?.length
            ? (
              <>
                { sectionTitle ? <SectionTitle>{sectionTitle}</SectionTitle> : null}
                <ContainerPosts>
                  {posts.map((post: PostOrPage, index: number) => {
                    return <GhostPostCard key={index} post={post} />
                  })}
                </ContainerPosts>
                {
                  coverSectionPageName &&
                  maxPaginationPages > 1
                    ? (
                        <Link href={`${coverSectionPageName}`} passHref>
                          <ShowMoreLink
                            className="show-more-link"
                          >
                            { t('posts_grid.show_more_link_text') }
                          </ShowMoreLink>
                        </Link>
                      )
                    : null}
              </>
              )
            : null
        }
      </Container>
    </ReactPlaceholder>
  )
}

export default React.memo(GhostPostsGrid)
