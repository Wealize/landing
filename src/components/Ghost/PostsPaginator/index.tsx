import React from 'react'
import Link from 'next/link'
import { Pagination } from '@tryghost/content-api'
import { useRouter } from 'next/router'

import { Container, PaginationPage } from './styles'


type GhostPostsPaginatorProps = {
  metaPagination: Pagination
}

const GhostPostsPaginator = (props: GhostPostsPaginatorProps) => {
  const { metaPagination } = props
  const router = useRouter()
  const {
    query: { section }
  } = router

  const pages = Array.from({ length: metaPagination?.pages }, (_, i) => i + 1)

  return (
    <Container>
      {pages?.map((page: number, index: number) =>
        <Link href={`${section}?page=${page}`} key={index} passHref>
          <PaginationPage className={`pagination-page-item${metaPagination?.page === page ? ' current' : ''}`} >
            {page}
          </PaginationPage>
        </Link>)}
    </Container>
  )
}

export default React.memo(GhostPostsPaginator)
