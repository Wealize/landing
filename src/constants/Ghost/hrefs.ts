export const CLIENT_STORY_SECTION = 'client-stories'
export const NEWS_ARTICLES_SECTION = 'news-articles'
