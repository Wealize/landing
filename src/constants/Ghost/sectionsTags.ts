export const NEWS_TAG_SLUG = 'news'
export const ARTICLE_TAG_SLUG = 'article'
export const CLIENT_STORY_TAG_SLUG = 'client-story'
export const EXTERNAL_CONTENT_TAG_SLUG = 'external-content'
