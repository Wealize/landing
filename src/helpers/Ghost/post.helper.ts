import { PostOrPage, Tag } from '@tryghost/content-api'

export const getSluggedTagsFromPost = (post: PostOrPage):string[] => {
  if (!post || !post?.tags) return []
  return post?.tags.map((tag: Tag) => tag.slug)
}
