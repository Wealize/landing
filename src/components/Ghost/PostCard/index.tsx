import React from 'react'
import Link from 'next/link'
import { PostOrPage } from '@tryghost/content-api'

import { ACCENT_COLOR, WHITE_COLOR } from '../../../theme/color'
import { isClientStory, isExternalContent, getFirstHref } from '../../../helpers/Ghost/post'
import ImageWithPlaceholder from '../../ImageWithPlaceholder'

import {
  Anchor,
  Article,
  Header,
  EntryContent,
  CardImage,
  CardHeadline,
  CardPublishedAt
} from './styles'

type GhostPostCardProps = {
  post: PostOrPage,
  color?: string,
  textColor?: string
}

const GhostPostCard = (props: GhostPostCardProps) => {
  const {
    post,
    color = WHITE_COLOR,
    textColor = ACCENT_COLOR
  } = props

  const isValidExternalContent = () => isExternalContent(post) && getFirstHref(post)

  return (
    <Link href={
      isValidExternalContent()
        ? getFirstHref(post)
        : `/news/${post.slug}`}
      passHref
    >
      <Anchor
        className={`${post.featured ? 'post-card' : 'post-card featured'}`}
        color={color}
        textColor={textColor}
        isFeatured={post?.featured}
        title={`postcard ${post.title}`}
        aria-label={`postcard ${post.title}`}
        target={isValidExternalContent() ? '_blank' : '_self'}
        {...(isValidExternalContent() ? { rel: 'noreferrer' } : {})}
      >
        <Article>
          <Header>
            <CardImage isVisible={post?.feature_image?.length > 0}>
            {post?.feature_image
              ? (
                  <ImageWithPlaceholder
                    imageUrl={post.feature_image}
                    imageAriaLabel={`post ${post.title}`}
                    imageTitle={`post ${post.title}`}
                    placeholderType='rect'
                    style={{ position: 'absolute', top: '0' }}
                  />
                )
              : null }
          </CardImage>
            <CardHeadline>{post.title}</CardHeadline>
          </Header>
          <EntryContent>
            <p>{post.custom_excerpt}</p>
            <CardPublishedAt isShowDate={!isClientStory(post)}>{post.published_at}</CardPublishedAt>
          </EntryContent>
        </Article>
      </Anchor>
    </Link>
  )
}

export default React.memo(GhostPostCard)
