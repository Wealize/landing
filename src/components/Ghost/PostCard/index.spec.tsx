import React from 'react'
import { mount } from 'enzyme'

import { WITHOUT_TAGS } from '../../../__fixtures__/Ghost/posts'

import GhostPostCard from './index'

describe('<GhostPostCard />', () => {
  const wrapper = mount(<GhostPostCard post={WITHOUT_TAGS} />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render content correctly', () => {
    expect(wrapper.find('a').length).toEqual(1)
    expect(wrapper.find('a').prop('href')).toContain(`/news/${WITHOUT_TAGS.slug}`)
    expect(wrapper.find('img').length).toEqual(1)
    expect(wrapper.find('p').text()).toEqual(WITHOUT_TAGS.custom_excerpt)
    expect(wrapper.find('h3').text()).toEqual(WITHOUT_TAGS.title)
    expect(wrapper.find('span').text()).toEqual('2021-02-01T11:00:20.000+00:00')

    const ghostPostCardDataWithoutFeatureImage = {
      ...WITHOUT_TAGS,
      feature_image: null
    }
    const wrapperWithoutFeatureImage = mount(<GhostPostCard post={ghostPostCardDataWithoutFeatureImage} />)
    expect(wrapperWithoutFeatureImage.find('img').length).toEqual(0)
  })
})
