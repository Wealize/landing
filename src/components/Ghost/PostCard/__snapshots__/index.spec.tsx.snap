// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GhostPostCard /> wrapper: should render 1`] = `
<Memo(GhostPostCard)
  post={
    Object {
      "canonical_url": null,
      "codeinjection_foot": null,
      "codeinjection_head": null,
      "comment_id": "6017df3f6b0d7a001c15cc7c",
      "created_at": "2021-02-01T11:00:15.000+00:00",
      "custom_excerpt": "Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.",
      "custom_template": null,
      "excerpt": "Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.",
      "feature_image": "https://static.ghost.org/v3.0.0/images/writing-posts-with-ghost.png",
      "featured": false,
      "html": "<h2 id=\\"just-start-writing\\">Just start writing</h2><p>Ghost has a powerful visual editor with familiar formatting options, as well as the ability to add dynamic content.</p><p>Select your text to add formatting such as headers or to create links. Or use Markdown shortcuts to do the work for you - if that's your thing. </p><figure class=\\"kg-card kg-image-card\\"><img src=\\"https://static.ghost.org/v2.0.0/images/formatting-editor-demo.gif\\" class=\\"kg-image\\"></figure><h2 id=\\"rich-editing-at-your-fingertips\\">Rich editing at your fingertips</h2><p>The editor can also handle rich media objects, called <strong>cards</strong>, which can be organised and re-ordered using drag and drop. </p><p>You can insert a card either by clicking the  <code>+</code>  button, or typing  <code>/</code>  on a new line to search for a particular card. This allows you to efficiently insert<strong> images</strong>, <strong>markdown</strong>, <strong>html, embeds </strong>and more.</p><p><strong>For example</strong>:</p><ul><li>Insert a video from YouTube directly by pasting the URL</li><li>Create unique content like buttons or forms using the HTML card</li><li>Need to share some code? Embed code blocks directly </li></ul><pre><code>&lt;header class=\\"site-header outer\\"&gt;
    &lt;div class=\\"inner\\"&gt;
        {{&gt; \\"site-nav\\"}}
    &lt;/div&gt;
&lt;/header&gt;</code></pre><p>It's also possible to share links from across the web in a visual way using bookmark cards that automatically render information from a websites meta data. Paste any URL to try it out: </p><figure class=\\"kg-card kg-bookmark-card\\"><a class=\\"kg-bookmark-container\\" href=\\"https://ghost.org\\"><div class=\\"kg-bookmark-content\\"><div class=\\"kg-bookmark-title\\">Ghost: The #1 open source headless Node.js CMS</div><div class=\\"kg-bookmark-description\\">The world’s most popular modern open source publishing platform. A headless Node.js CMS used by Apple, Sky News, Tinder and thousands more. MIT licensed, with 30k+ stars on Github.</div><div class=\\"kg-bookmark-metadata\\"><img class=\\"kg-bookmark-icon\\" src=\\"https://ghost.org/icons/icon-512x512.png?v=188b8b6d743c6338ba2eab2e35bab4f5\\"><span class=\\"kg-bookmark-publisher\\">Ghost</span></div></div><div class=\\"kg-bookmark-thumbnail\\"><img src=\\"https://ghost.org/images/meta/Ghost.png\\"></div></a></figure><h2 id=\\"working-with-images-in-posts\\">Working with images in posts</h2><p>You can add images to your posts in many ways:</p><ul><li>Upload from your computer</li><li>Click and drag an image into the browser</li><li>Paste directly into the editor from your clipboard</li><li>Insert using a URL</li></ul><h3 id=\\"image-sizes\\">Image sizes</h3><p>Once inserted you can blend images beautifully into your content at different sizes and add captions and alt tags wherever needed.</p><figure class=\\"kg-card kg-image-card\\"><img src=\\"https://static.ghost.org/v3.0.0/images/image-sizes-ghost-editor.png\\" class=\\"kg-image\\"></figure><h3 id=\\"image-galleries\\">Image galleries</h3><p>Tell visual stories using the gallery card to add up to 9 images that will display as a responsive image gallery: </p><figure class=\\"kg-card kg-gallery-card kg-width-wide\\"><div class=\\"kg-gallery-container\\"><div class=\\"kg-gallery-row\\"><div class=\\"kg-gallery-image\\"><img src=\\"https://static.ghost.org/v3.0.0/images/gallery-sample-1.jpg\\" width=\\"6000\\" height=\\"4000\\"></div><div class=\\"kg-gallery-image\\"><img src=\\"https://static.ghost.org/v3.0.0/images/gallery-sample-2.jpg\\" width=\\"5746\\" height=\\"3831\\"></div><div class=\\"kg-gallery-image\\"><img src=\\"https://static.ghost.org/v3.0.0/images/gallery-sample-3.jpg\\" width=\\"5872\\" height=\\"3915\\"></div></div></div></figure><h3 id=\\"image-optimisation\\">Image optimisation</h3><p>Ghost will automatically resize and optimise your images with lossless compression. Your posts will be fully optimised for the web without any extra effort on your part.</p><h2 id=\\"next-publishing-options\\">Next: Publishing Options</h2><p>Once your post is looking good, you'll want to use the <a href=\\"https://wlz-blog.herokuapp.com/publishing-options/\\">publishing options</a> to ensure it gets distributed in the right places, with custom meta data, feature images and more.</p>",
      "id": "6017df3f6b0d7a001c15cc7c",
      "meta_description": null,
      "meta_title": null,
      "og_description": null,
      "og_image": null,
      "og_title": null,
      "published_at": "2021-02-01T11:00:20.000+00:00",
      "reading_time": 3,
      "slug": "the-editor",
      "title": "Writing posts with Ghost ✍️",
      "twitter_description": null,
      "twitter_image": null,
      "twitter_title": null,
      "updated_at": "2021-02-01T11:00:15.000+00:00",
      "url": "https://wlz-blog.herokuapp.com/the-editor/",
      "uuid": "b4283683-b51e-4c67-ba42-66de093b1e05",
    }
  }
>
  <Link
    href="/news/the-editor"
    passHref={true}
  >
    <styles__Anchor
      aria-label="postcard Writing posts with Ghost ✍️"
      className="post-card featured"
      color="#ffffff"
      href="/news/the-editor"
      isFeatured={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      target="_self"
      textColor="#6666ff"
      title="postcard Writing posts with Ghost ✍️"
    >
      <a
        aria-label="postcard Writing posts with Ghost ✍️"
        className="styles__Anchor-pmxezz-0 goXYDO post-card featured"
        color="#ffffff"
        href="/news/the-editor"
        onClick={[Function]}
        onMouseEnter={[Function]}
        target="_self"
        title="postcard Writing posts with Ghost ✍️"
      >
        <styles__Article>
          <article
            className="styles__Article-pmxezz-1 kulkFr"
          >
            <styles__Header>
              <header
                className="styles__Header-pmxezz-2"
              >
                <styles__CardImage
                  isVisible={true}
                >
                  <div
                    className="styles__CardImage-pmxezz-3 loizaj"
                  >
                    <Memo(ImageWithPlaceholder)
                      imageAriaLabel="post Writing posts with Ghost ✍️"
                      imageTitle="post Writing posts with Ghost ✍️"
                      imageUrl="https://static.ghost.org/v3.0.0/images/writing-posts-with-ghost.png"
                      placeholderType="rect"
                      style={
                        Object {
                          "position": "absolute",
                          "top": "0",
                        }
                      }
                    >
                      <styles__Container
                        style={
                          Object {
                            "position": "absolute",
                            "top": "0",
                          }
                        }
                      >
                        <div
                          className="styles__Container-sc-1li6v9v-0 HMjkw"
                          style={
                            Object {
                              "position": "absolute",
                              "top": "0",
                            }
                          }
                        >
                          <ReactPlaceholder
                            firstLaunchOnly={true}
                            ready={false}
                            showLoadingAnimation={true}
                            style={
                              Object {
                                "marginRight": "0",
                                "position": "absolute",
                              }
                            }
                            type="rect"
                          >
                            <RectShape
                              className="show-loading-animation"
                              color="#CDCDCD"
                              rows={3}
                              style={
                                Object {
                                  "marginRight": "0",
                                  "position": "absolute",
                                }
                              }
                            >
                              <div
                                className="rect-shape show-loading-animation"
                                style={
                                  Object {
                                    "backgroundColor": "#CDCDCD",
                                    "height": "100%",
                                    "marginRight": "0",
                                    "position": "absolute",
                                    "width": "100%",
                                  }
                                }
                              />
                            </RectShape>
                          </ReactPlaceholder>
                          <Image
                            aria-label="post Writing posts with Ghost ✍️"
                            layout="fill"
                            objectFit="cover"
                            onLoad={[Function]}
                            quality={75}
                            src="https://static.ghost.org/v3.0.0/images/writing-posts-with-ghost.png"
                            title="post Writing posts with Ghost ✍️"
                          >
                            <div
                              style={
                                Object {
                                  "bottom": 0,
                                  "boxSizing": "border-box",
                                  "display": "block",
                                  "left": 0,
                                  "margin": 0,
                                  "overflow": "hidden",
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                }
                              }
                            >
                              <img
                                aria-label="post Writing posts with Ghost ✍️"
                                decoding="async"
                                onLoad={[Function]}
                                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                                style={
                                  Object {
                                    "border": "none",
                                    "bottom": 0,
                                    "boxSizing": "border-box",
                                    "display": "block",
                                    "height": 0,
                                    "left": 0,
                                    "margin": "auto",
                                    "maxHeight": "100%",
                                    "maxWidth": "100%",
                                    "minHeight": "100%",
                                    "minWidth": "100%",
                                    "objectFit": "cover",
                                    "objectPosition": undefined,
                                    "padding": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "visibility": "hidden",
                                    "width": 0,
                                  }
                                }
                                title="post Writing posts with Ghost ✍️"
                              />
                            </div>
                          </Image>
                        </div>
                      </styles__Container>
                    </Memo(ImageWithPlaceholder)>
                  </div>
                </styles__CardImage>
                <styles__CardHeadline>
                  <h3
                    className="styles__CardHeadline-pmxezz-4 fpUjLl"
                  >
                    Writing posts with Ghost ✍️
                  </h3>
                </styles__CardHeadline>
              </header>
            </styles__Header>
            <styles__EntryContent>
              <div
                className="styles__EntryContent-pmxezz-5 fbeNuV"
              >
                <p>
                  Discover familiar formatting options in a functional toolbar and the ability to add dynamic content seamlessly.
                </p>
                <styles__CardPublishedAt
                  isShowDate={true}
                >
                  <span
                    className="styles__CardPublishedAt-pmxezz-6 bjlrxv"
                  >
                    2021-02-01T11:00:20.000+00:00
                  </span>
                </styles__CardPublishedAt>
              </div>
            </styles__EntryContent>
          </article>
        </styles__Article>
      </a>
    </styles__Anchor>
  </Link>
</Memo(GhostPostCard)>
`;
